pkg/operator/ceph/pool/controller.go

```diff
@@ -334,6 +339,8 @@ func (r *ReconcileCephBlockPool) reconcile(request reconcile.Request) (reconcile
 		}
 	}
 
+	// update ObservedGeneration in status at the end of reconcile
+	r.updateObservedGenerationInStatus(request.NamespacedName, observedGeneration)
```

Comment:
Updated observedGeneration separately, because passing cephBlockPool current object on `updateStatus()` function was failing.

Reply:
Instead of updating the observed generation separately, let's pass the observedGeneration as a parameter to `UpdateStatus()`. It would only be set at the end of the reconcile, so any other calls to `UpdateStatus()` should preserve the existing observedGeneration that might have been set on a previous reconcile.

Reply:
Yes right, the thing that is making me confused is what should be the `status type(status cephv1.ConditionType)` I should pass because there is no calling of updatestatus() function at the end of reconcile.